settings.php

```diff
@@ -25,23 +25,98 @@
+        new lang_string('settings:enablelogging', 'tool_objectfs'), '', '', '0', '1'));
+
+
+    $settings->add(new admin_setting_heading('tool_objectfs/filetransfersettings',
+        new lang_string('settings:filetransferheader', 'tool_objectfs'), ''));
+
+    $settings->add(new admin_setting_configtext('tool_objectfs/sizethreshold',
+        new lang_string('settings:sizethreshold', 'tool_objectfs'), '', 1024 * 10, PARAM_INT));
+
+    $settings->add(new admin_setting_configtext('tool_objectfs/batchsize',
+        new lang_string('settings:batchsize', 'tool_objectfs'), '', 10000, PARAM_INT));
+
+    $settings->add(new admin_setting_configduration('tool_objectfs/minimumage',
+        new lang_string('settings:minimumage', 'tool_objectfs'), '', '', 7 * DAYSECS));
+
+    $settings->add(new admin_setting_configcheckbox('tool_objectfs/deletelocal',
+        new lang_string('settings:deletelocal', 'tool_objectfs'), '', '', '0', '1'));
+
+    $settings->add(new admin_setting_configduration('tool_objectfs/consistencydelay',
+        new lang_string('settings:consistencydelay', 'tool_objectfs'), '', '', 10 * MINSECS));
+
+
+    $settings->add(new admin_setting_heading('tool_objectfs/storagefilesystemselection',
+        new lang_string('settings:clientselection:header', 'tool_objectfs'), ''));
+
+    $settings->add(new admin_setting_configselect('tool_objectfs/filesystem',
+        new lang_string('settings:clientselection:title', 'tool_objectfs'),
+        new lang_string('settings:clientselection:title_help', 'tool_objectfs'), '',
+        tool_objectfs_get_fs_list()));
+
+
+    $config = get_objectfs_config();
+
+    if (tool_objectfs_filesystem_supports_presigned_urls($config->filesystem)) {
+        $settings->add(new admin_setting_heading('tool_objectfs/presignedurls',
+            new lang_string('settings:presignedurl:header', 'tool_objectfs'), ''));
+
+        $settings->add(new admin_setting_configcheckbox('tool_objectfs/enablepresignedurls',
+            new lang_string('settings:presignedurl:enablepresignedurls', 'tool_objectfs'),
+            new lang_string('settings:presignedurl:enablepresignedurls_help', 'tool_objectfs'), '', '0', '1'));
+
+        $settings->add(new admin_setting_configduration('tool_objectfs/expirationtime',
+            new lang_string('settings:presignedurl:expirationtime', 'tool_objectfs'),
+            new lang_string('settings:presignedurl:expirationtime_help', 'tool_objectfs'), '', 10 * MINSECS));
+
+        $settings->add(new admin_setting_configtext('tool_objectfs/presignedminfilesize',
+            new lang_string('settings:presignedurl:presignedminfilesize', 'tool_objectfs'),
+            new lang_string('settings:presignedurl:presignedminfilesize_help', 'tool_objectfs'), 0, PARAM_INT));
+    }
+
+    $client = tool_objectfs_get_client($config);
+    if ($client && $client->get_availability() && $PAGE->url->compare(new moodle_url('/admin/settings.php?section=tool_objectfs'))) {
```

Comment:
This feels like it should be split in two, if the client is available then I think you should show define_client_section

But then inside that, only if you are in ?section=tool_objectfs should you do the extra step of the 'define_client_check'

ie I think define_client_check needs to be called from settings.php and have a consistent api, ie rename define_amazon_s3_check

Reply:
What about the argument supplied to define_amazon_s3_check? Remove it and the condition check on line 250? Or specifically check in settings.php if it is a DO filesystem and then supply false in the arguments?

Reply:
might be worth the DO one overriding it and adding the second check instead of one shared method with a code fork